### PR TITLE
fix test_utils tests

### PR DESCRIPTION
**Pull request by ahartikainen**
#### Submission Checklist

- [ ] Run unit tests
- [ ] Declare copyright holder and open-source license: see below

#### Summary

I don't know a good way to test multiple cmdstan versions without installing multiple versions.
Also changed a bunch of tests to use tempfolder.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Ari Hartikainen

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by ahartikainen** [comment]
This should also fix other currently failing tests, that fail due to change of CMDSTAN env var (there was an exception before the value was changed back to the correct one)

**Comment by mitzimorris** [comment]
are these tests fundamentally too brittle to maintain?

**Comment by ahartikainen** [comment]
Some might be. Do we want tests to depend on cmdstan? Some of them must, but maybe some of the tests could use static files?

Feel free to edit the PR.

**Comment by ahartikainen** [comment]
I think install_cmdstan is broken for Windows. I need to investigate.

**Comment by mitzimorris** [comment]
hi @ahartikainen - CmdStanR users have also been unable to install cmdstan on Windows - your insight on what's causing this would be much appreciated.

I have a brand-new laptop running Windows 10 Pro - investigating here too.

**Comment by ahartikainen** [comment]
I think `cmdstan_version_at(2, 25)` should catch exceptions and just return False?

**Comment by ahartikainen** [comment]
Ok, let's see

I update some install_cmdstan stuff, now it installs on appveyor.
I added try-except to cmdstan_version_at

If tests pass, let's think what to do with this PR. (either use this or create a new one to do better fixes)

**Comment by ahartikainen** [comment]
Ok, I think new cmdstan outputs more lines than before and if Windows stdout/stderr (?) buffer is not emptied... it might get stuck. This also probably happens in R too.

I have no idea what is going on, but without `--verbose` flag, install_cmdstan fails on my computer. With `--verbose` everything works. (I think at least `boost` has a new output look)

**Comment by mitzimorris** [comment]
running `install_cmdstan` right now on my laptop.  it's taking forever...

File: test/test_utils.py [hunk 1]
```diff
@@ -145,41 +148,42 @@ def test_validate_path(self):
...
     def test_cmdstan_version_at(self):
         cmdstan_path()  # sets os.environ['CMDSTAN']
-        good_path = os.environ['CMDSTAN']
-        del os.environ['CMDSTAN']
-        os.environ['CMDSTAN'] = os.path.join(
-            os.path.dirname(good_path), 'cmdstan-2.24.1'
-        )
-        self.assertTrue(cmdstan_version_at(2, 24))
-        self.assertFalse(cmdstan_version_at(2, 25))
+        self.assertTrue(cmdstan_version_at(2, 25))
```
Comment (mitzimorris) [inline test/test_utils.py:186]: this is still too brittle - how about:
`self.assertFalse(cmdstan_version_at(99,99)`

**Comment by ahartikainen** [comment]
Hmm, I think we also need different numbers in Windows.

What if we test first 10 nums and then the lenght --> 18 digits?

**Comment by ahartikainen** [comment]
Did you use `--verbose`

**Comment by ahartikainen** [comment]
Ok, tests are passed (travis has stop now OSS CI, and one of our jobs is "still" running).

**Comment by mitzimorris** [comment]
@serban-nicusor-toptal - what's up with Travis?  it's showing message about negative credit balance -
Untitled

**Comment by serban-nicusor-toptal** [comment]
Hey @mitzimorris I did not setup Travis nor have access to the plans page to see what's wrong. @rok-cesnovar do you know by any chance who setup Travis ?

**Comment by rok-cesnovar** [comment]
To my knowledge, at least since I have been more involved, the Stan project never had any paid plans for Travis.

Travis did however limit free cycles for projects. So I would say `cmdstanpy` repo ran out of free cycles for November.

**Comment by ahartikainen** [comment]
https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

**Comment by mitzimorris** [comment]
is this PR blocking https://github.com/stan-dev/cmdstanpy/pull/329?
if so, can we merge on the grounds that tests on Python 3.6 and 3.7 passed, and Travis timed us out?
alternative is wait 72 more hours.

**Pull request merged**